fuzzers/qemu_launcher/src/harness.rs

```diff
@@ -0,0 +1,85 @@
+use libafl::{
+    executors::ExitKind,
+    inputs::{BytesInput, HasTargetBytes},
+    Error,
+};
+use libafl_bolts::AsSlice;
+use libafl_qemu::{ArchExtras, CallingConvention, Emulator, GuestAddr, GuestReg, MmapPerms, Regs};
+
+pub struct Harness<'a> {
+    emu: &'a Emulator,
+    input_addr: GuestAddr,
+    pc: GuestAddr,
+    stack_ptr: GuestAddr,
+    ret_addr: GuestAddr,
+}
+
+pub const MAX_INPUT_SIZE: usize = 1048576; // 1MB
+
+impl<'a> Harness<'a> {
+    pub fn new(emu: &Emulator) -> Result<Harness, Error> {
+        let input_addr = emu
+            .map_private(0, MAX_INPUT_SIZE, MmapPerms::ReadWrite)
+            .map_err(|e| Error::unknown(format!("Failed to map input buffer: {e:}")))?;
+
+        let pc: GuestReg = emu
+            .read_reg(Regs::Pc)
+            .map_err(|e| Error::unknown(format!("Failed to read PC: {e:}")))?;
+
+        let stack_ptr: GuestAddr = emu
+            .read_reg(Regs::Sp)
+            .map_err(|e| Error::unknown(format!("Failed to read stack pointer: {e:}")))?;
+
+        let ret_addr: GuestAddr = emu
+            .read_return_address()
+            .map_err(|e| Error::unknown(format!("Failed to read return address: {e:}")))?;
+
+        Ok(Harness {
+            emu,
+            input_addr,
+            pc,
+            stack_ptr,
+            ret_addr,
+        })
+    }
+
+    pub fn run(&self, input: &BytesInput) -> ExitKind {
+        self.reset(input).unwrap();
+        ExitKind::Ok
+    }
+
+    fn reset(&self, input: &BytesInput) -> Result<(), Error> {
+        let target = input.target_bytes();
+        let mut buf = target.as_slice();
+        let mut len = buf.len();
+        if len > MAX_INPUT_SIZE {
+            buf = &buf[0..MAX_INPUT_SIZE];
+            len = MAX_INPUT_SIZE;
+        }
+        let len = len as GuestReg;
+
+        unsafe { self.emu.write_mem(self.input_addr, buf) };
+
+        self.emu
+            .write_reg(Regs::Pc, self.pc)
+            .map_err(|e| Error::unknown(format!("Failed to write PC: {e:}")))?;
```

Comment:
I love your detailed error handling (and please do leave it that way), but for the future, probably a `?` could be good enough here? Not sure how verbose the emu error messages are, but maybe we should make hem more verbose if they are not sufficient

Reply:
I think I needed to convert the error type maybe?